pkg/manager/member/tidb_member_manager.go

```diff
@@ -355,6 +369,96 @@ func (tmm *tidbMemberManager) syncTiDBService(tc *v1alpha1.TidbCluster) error {
 	return nil
 }
 
+// syncTiDBConfigMap syncs the configmap of tidb
+func (tmm *tidbMemberManager) syncTiDBConfigMap(tc *v1alpha1.TidbCluster, set *apps.StatefulSet) (*corev1.ConfigMap, error) {
+
+	// For backward compatibility, only sync tidb configmap when .tidb.config is non-nil
+	if tc.Spec.TiDB.Config == nil {
+		return nil, nil
+	}
+	newCm, err := getTiDBConfigMap(tc)
+	if err != nil {
+		return nil, err
+	}
+	if set != nil && tc.TiDBConfigUpdateStrategy() == v1alpha1.ConfigUpdateStrategyInPlace {
+		inUseName := FindConfigMapVolume(&set.Spec.Template.Spec, func(name string) bool {
+			return strings.HasPrefix(name, controller.TiDBMemberName(tc.Name))
+		})
+		if inUseName != "" {
+			newCm.Name = inUseName
+		}
+	}
+
+	return tmm.typedControl.CreateOrUpdateConfigMap(tc, newCm)
+}
+
+func getTiDBConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
+
+	config := tc.Spec.TiDB.Config
+	if config == nil {
+		return nil, nil
+	}
+
+	// override CA if tls enabled
+	if tc.Spec.EnableTLSCluster {
+		if config.Security == nil {
+			config.Security = &v1alpha1.Security{}
+		}
+		config.Security.ClusterSSLCA = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
```

Comment:
It's better to define these 3 paths as constant.

Reply:
Good catch!